apps/nsqd/nsqd.go

```diff
@@ -88,6 +88,8 @@ func nsqFlagset() *flag.FlagSet {
 
 	authHTTPAddresses := app.StringArray{}
 	flagSet.Var(&authHTTPAddresses, "auth-http-address", "<addr>:<port> to query auth server (may be given multiple times)")
+	flagSet.Bool("http-auth-required", false, "require auth for HTTP clients")
+	flagSet.Bool("https-auth-required", false, "require auth for HTTPS clients")
```

Comment:
Agreed. Originally I was mirroring the `--*-address` flags.

One option might be to have an `--auth-required` or `--require-auth` flag. It could be specified multiple times, or it could be of the form `tcp,http,https` with a default of `tcp` to preserve existing functionality. The comma separated options breaks convention with existing flags, however without it you wouldn't be able to both preserve existing functionality and allow disabling auth for `tcp` without some funky logic.

What do you think?